test/run/ok/array.tc.ok

```diff
@@ -0,0 +1,3 @@
+array.mo:89.27-89.64: warning [M0074], this array has type
+  [Any]
+because elements have inconsistent types
```

Comment:
Interesting that the compiler still warns about it using this type `[Any]` even though the program has an explicit annotation saying that "it knows" that already.

Reply:
Yeah, that's why I filed #3302!